packages/core/test/financial-templates/contract-for-difference/ContractForDifferenceCreator.js

```diff
@@ -0,0 +1,197 @@
+const { toWei, utf8ToHex, hexToUtf8 } = web3.utils;
+const truffleAssert = require("truffle-assertions");
+const { assert } = require("chai");
+
+// Libraries and helpers
+const { interfaceName, didContractThrow, ZERO_ADDRESS } = require("@uma/common");
+
+// Tested Contract
+const ContractForDifference = artifacts.require("ContractForDifference");
+const ContractForDifferenceCreator = artifacts.require("ContractForDifferenceCreator");
+const ContractForDifferenceFinancialProjectLibraryTest = artifacts.require(
+  "ContractForDifferenceFinancialProjectLibraryTest"
+);
+
+// Helper contracts
+const AddressWhitelist = artifacts.require("AddressWhitelist");
+const IdentifierWhitelist = artifacts.require("IdentifierWhitelist");
+const Finder = artifacts.require("Finder");
+const Timer = artifacts.require("Timer");
+const OptimisticOracle = artifacts.require("OptimisticOracle");
+const Token = artifacts.require("ExpandedERC20");
+const TokenFactory = artifacts.require("TokenFactory");
+
+let collateralToken;
+let contractForDifferenceLibrary;
+let contractForDifferenceCreator;
+let collateralWhitelist;
+let identifierWhitelist;
+let optimisticOracle;
+let finder;
+let timer;
+let constructorParams;
+
+const startTimestamp = Math.floor(Date.now() / 1000);
+const expirationTimestamp = startTimestamp + 10000;
+const optimisticOracleLiveness = 7200;
+const priceFeedIdentifier = utf8ToHex("TEST_IDENTIFIER");
+const collateralPerPair = toWei("1"); // each pair of long and short tokens need 1 unit of collateral to mint.
+const syntheticName = "Test CFD";
+const syntheticSymbol = "tCFD";
+
+contract("ContractForDifferenceCreator", function (accounts) {
+  const deployer = accounts[0];
+  const sponsor = accounts[1];
+
+  before(async () => {
+    finder = await Finder.deployed();
+    timer = await Timer.deployed();
+    collateralWhitelist = await AddressWhitelist.deployed();
+
+    identifierWhitelist = await IdentifierWhitelist.deployed();
+    await identifierWhitelist.addSupportedIdentifier(priceFeedIdentifier, { from: deployer });
+  });
+
+  beforeEach(async function () {
+    // Force each test to start with a simulated time that's synced to the startTimestamp.
+    await timer.setCurrentTime(startTimestamp);
+
+    collateralToken = await Token.new("Wrapped Ether", "WETH", 18, { from: deployer });
+    await collateralToken.addMember(1, deployer, { from: deployer });
+    await collateralToken.mint(sponsor, toWei("1000"), { from: deployer });
+
+    await collateralWhitelist.addToWhitelist(collateralToken.address);
+
+    optimisticOracle = await OptimisticOracle.new(optimisticOracleLiveness, finder.address, timer.address);
```

Comment:
Just out of curiosity, why do we need to create a new OO? Isn't this provided by migrations?

Reply:
this is true. this mistake is actually repeated in a few CFD unit tests. I will update all these in a separate cleanup PR.